# HG changeset patch
# User Rik <octave@nomad.inbox5.com>
# Date 1313731396 25200
#      Thu Aug 18 22:23:16 2011 -0700
# Node ID f89449967f0d1020b12b015c681b62d73667283f
# Parent  6590446c2498ab12e775bfeaa06d34d6cfccbd04
deblank.m, strtrim.m: Correctly validate cellstr inputs (Bug #34038)

deblank.m, strtrim.m: Use iscellstr() instead of iscell() to validate input.

diff --git a/scripts/strings/deblank.m b/scripts/strings/deblank.m
--- a/scripts/strings/deblank.m
+++ b/scripts/strings/deblank.m
@@ -37,17 +37,17 @@ function s = deblank (s)
 
     k = find (! isspace (s) & s != "\0");
     if (isempty (s) || isempty (k))
       s = "";
     else
       s = s(:,1:ceil (max (k) / rows (s)));
     endif
 
-  elseif (iscell (s))
+  elseif (iscellstr (s))
 
     s = regexprep (s, "[\\s\v\\0]+$", '');
 
   else
     error ("deblank: S argument must be a string or cellstring");
   endif
 
 endfunction
diff --git a/scripts/strings/strtrim.m b/scripts/strings/strtrim.m
--- a/scripts/strings/strtrim.m
+++ b/scripts/strings/strtrim.m
@@ -48,17 +48,17 @@ function s = strtrim (s)
 
     k = find (! isspace (s));
     if (isempty (s) || isempty (k))
       s = "";
     else
       s = s(:, ceil (min (k) / rows (s)):ceil (max (k) / rows (s)));
     endif
 
-  elseif (iscell (s))
+  elseif (iscellstr (s))
 
     s = regexprep (s, "^[\\s\v]+|[\\s\v]+$", '');
 
   else
     error ("strtrim: S argument must be a string or cellstring");
   endif
 
 endfunction
